### DISCUSSION TITLE
Difficulties with installation instructions (beginner)

### DISCUSSION THREAD

**Discussion by CreativeLibre** [tbabej/taskwiki#332] [Community support]
As a relative newbie, I'm sorry to say that the installation instructions are absolutely poor. I unfortunately do not get it to run. In the current instruction so much is assumed and expected, but that does not apply to beginners. Is there somewhere a detailed tutorial on how I can install this package?

**Comment by tbabej** [comment]
Sorry you're having trouble! I agree the instructions are old-ish and maybe could use some polishing. In the meantime, is there a specific problem we can help you with?

**Comment by CreativeLibre** [comment]
Thanks for your answer. I will write the next few days all the problems I had during the installation together and report here.

**Comment by amagee** [comment]
I am not a beginner but I was also unable to get it to run.

Here's what I did:

- Installed taskwarrior v2.5.1 (`sudo apt install taskwarrior` on my Ubuntu machine)
- Installed vimwiki and taskwiki via Plug:

```
Plug 'vimwiki/vimwiki', { 'branch': 'dev' }
Plug 'tools-life/taskwiki'
```

- Installed the `develop` branch of the `tasklib` Python package into the Python virtual environment that's referred to in my `g:python_host_prog` and `g:python3_host_prog` Neovim variables:

```
pyenv activate <my neovim python env>
git clone git@github.com:tbabej/tasklib.git
cd tasklib
git checkout develop
pip install -e .
```

Note: the instructions suggest `sudo pip3 install --upgrade -r requirements.txt` but there is no requirements.txt file in the linked repo.

I receive no error messages and the all the packages are installed (eg. `:help taskwiki` works) but the desired behaviour: "Such tasks get synced to TaskWarrior when the file is saved" does not occur.

Ie. if I open `~/.vimwiki/index.wiki` and enter 

```
* [ ] Install Taskwiki
```

and save, nothing happens to my Taskwarrior files.

**Comment by amagee** [comment]
Update: I noticed there was a `requirements.txt` file in the taskwiki checkout itself, and installing from that instead of cloning the develop branch of `tasklib` made things work.

**Reply by mukundraj** [reply]
I had same problem of tasks not showing up in taskwarrior. This solution (installing tasklib using the requirements.txt) worked for me too!

**Comment by MaxPancake** [comment]
Beginner + Mac user - been using vimwiki and task warrior separately, but can't get taskwiki to work. Tried adding tasks in vimwiki, but nothing happened..

**Reply by MaxPancake** [reply]
Solved my own problem.. Turned out I installed vimwiki and taskwiki the wrong way. Uninstalled both and did a clean install. Now got taskwiki working -- for most features, but not all.. Still can't get metadata (using "--") to work somehow??
